givenergy_modbus/model/__init__.py

```diff
@@ -60,4 +60,8 @@ def from_repr(cls, start: int | str, end: int | str):
             end = f"{end:04d}"
         end_hour = int(end[:-2])
         end_minute = int(end[-2:])
-        return cls(time(start_hour, start_minute), time(end_hour, end_minute))
+        try:
+            return cls(time(start_hour, start_minute), time(end_hour, end_minute))
+        except:
+            # if there's garbage data return midnight
+            return cls(time(0,0), time(0,0))
```

Comment:
I generally prefer to make a fuss (throw an exception) when given garbage, rather than silenty do something else. Since I'd want to highlight to the user that they've done something stupid.  But if this is also used for converting registers read from the inverter, we do need to be a bit more tolerant.

Reply:
Yeah, as it’s hidden in the register get it throws an exception and causes problems, other option is to return none?

Reply:
> other option is to return none?

My gut feeling would be to prefer that, though of course you have much more experience with actually using this software.

I changed the default for any unavailable register from 0 to None, since it's just too confusing to be told a value is 0 when in fact it's unknown.